backend/kesaseteli/applications/views.py

```diff
@@ -163,3 +170,123 @@ def export_and_download_annual_applications(
             return self.render_error(_("Hakemuksia ei löytynyt."))
 
         return self.get_xlsx_response(queryset, columns)
+
+
+class YouthApplicationExcelExportView(APIView):
+    permission_classes = [AllowAny]  # Permissions are handled per function
+    http_method_names = ["get", "head", "options"]
+    serializer_class = YouthApplicationExcelExportSerializer
+
+    @staticmethod
+    def source_fields_and_output_names() -> List[Tuple[str, str]]:
+        with translation.override("fi"):
+            return [
+                ("application_date", str(_("Hakupvm"))),
+                ("id", str(_("Tunniste"))),
+                ("status", str(_("Hakemuksen tila"))),
+                ("application_year", str(_("Hakuvuosi"))),
+                ("summer_voucher_serial_number", str(_("Kesäsetelinro"))),
+                ("birth_year", str(_("Syntymävuosi"))),
+                ("birthdate", str(_("Syntymäpvm"))),
+                ("first_name", str(_("Etunimi"))),
+                ("last_name", str(_("Sukunimi"))),
+                ("vtj_last_name", str(_("VTJ-sukunimi"))),
+                ("school", str(_("Koulu"))),
+                ("is_unlisted_school", str(_("Listaamaton koulu?"))),
+                ("email", str(_("Sähköposti"))),
+                ("phone_number", str(_("Puhelinnro"))),
+                ("postcode", str(_("Postinro"))),
+                ("vtj_home_municipality", str(_("VTJ-kotikunta"))),
+                ("additional_info_user_reasons", str(_("Lisätietosyyt"))),
+                ("additional_info_description", str(_("Lisätiedot"))),
+                ("language", str(_("Kieli"))),
+                ("confirmation_date", str(_("Vahvistettu"))),
+                ("additional_info_providing_date", str(_("Lisätiedot annettu"))),
+                ("handling_date", str(_("Käsitelty"))),
+            ]
+
+    @staticmethod
+    def source_fields() -> Tuple[str, ...]:
+        return list(
+            zip(*YouthApplicationExcelExportView.source_fields_and_output_names())
+        )[0]
+
+    @staticmethod
+    def output_column_names() -> Tuple[str, ...]:
+        return list(
+            zip(*YouthApplicationExcelExportView.source_fields_and_output_names())
+        )[1]
+
+    def get_queryset(self):
+        return YouthApplication.objects.active().order_by("created_at", "pk")
+
+    def serializer(self, apps: QuerySet[YouthApplication]):
+        return (self.generate_data_row(app) for app in apps)
+
+    @enforce_handler_view_adfs_login
+    def get(
+        self, request, *args, **kwargs
+    ) -> Union[HttpResponse, HttpResponseRedirect, StreamingHttpResponse]:
+        queryset = self.get_queryset()
```

Comment:
Should we log to audit log about who has downloaded the excel especially because it has sensitive information about the youth persons?

Reply:
Not sure if the logging should be here or `urls.py`

Reply:
That's a legitimate point, I'll see about how to add some audit logging here...

Reply:
Added audit logging